File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer> array=new HashMap<>();
        int n=nums.length;
        for(int i:nums)
        {
            array.put(i,array.getOrDefault(i,0)+1);
            if(array.get(i)>(n/2))
            return i;
        }
        return -1;
    }
}

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n=len(nums)
        s=dict()
        for i in nums:
            a=s.get(i,0)+1
            if(a>=(n/2)):
                return i
            s[i]=a
        return -1